webapp/src/components/map/map-slice.tsx

```diff
@@ -0,0 +1,1576 @@
+
+  const filteredStartTime = startTime + sliderValue / 10 - timeWindowSeconds
+  const filteredEndTime = startTime + sliderValue / 10
+
+  return [filteredStartTime, filteredEndTime]
+}
+
+const _updateQueryParams = ({
+  state,
+  startDate,
+  endDate,
+  eventDate,
+  vehicleId,
+  intersectionId,
+  roadRegulatorId,
+  resetTimeWindow,
+  updateSlider,
+}: {
+  state: RootState['intersectionMap']
+  startDate?: Date
+  endDate?: Date
+  eventDate?: Date
+  vehicleId?: string
+  intersectionId?: number
+  roadRegulatorId?: number
+  resetTimeWindow?: boolean
+  updateSlider?: boolean
+}) => {
+  const newQueryParams = {
+    startDate: startDate ?? state.value.queryParams.startDate,
+    endDate: endDate ?? state.value.queryParams.endDate,
+    eventDate: eventDate ?? state.value.queryParams.eventDate,
+    vehicleId: vehicleId ?? state.value.queryParams.vehicleId,
+    intersectionId: intersectionId ?? state.value.queryParams.intersectionId,
+    roadRegulatorId: roadRegulatorId ?? state.value.queryParams.roadRegulatorId,
+  }
+  if (compareQueryParams(state.value.queryParams, newQueryParams)) {
+    state.value.queryParams = newQueryParams
+    state.value.sliderTimeValue = getNewSliderTimeValue(
+      state.value.queryParams.startDate,
+      state.value.sliderValue,
+      state.value.timeWindowSeconds
+    )
+    if (resetTimeWindow) state.value.timeWindowSeconds = 60
+    if (updateSlider) state.value.sliderValue = getTimeRange(newQueryParams.startDate, newQueryParams.endDate)
+  }
+}
+
+export const downloadMapData = createAsyncThunk(
+  'intersectionMap/downloadMapData',
+  async (_, { getState }) => {
+    const currentState = getState() as RootState
+    const rawData = selectRawData(currentState)!
+    const queryParams = selectQueryParams(currentState)
+
+    return downloadAllData(rawData, queryParams)
+  },
+  {
+    condition: (_, { getState }) =>
+      selectToken(getState() as RootState) != undefined &&
+      selectQueryParams(getState() as RootState).intersectionId != undefined &&
+      selectQueryParams(getState() as RootState).roadRegulatorId != undefined,
+  }
+)
+
+export const intersectionMapSlice = createSlice({
+  name: 'intersectionMap',
+  initialState: {
+    loading: false,
+    value: initialState,
+  },
+  reducers: {
+    updateSsmSrmCounts: (state, action: PayloadAction<string>) => {
+      // let localSrmCount = 0
+      // let localSsmCount = 0
+      // let localMsgList = []
+      // // console.error('srmSsmList', state.value.srmSsmList)
+      // for (const elem of state.value.srmSsmList) {
+      //   if (elem.ip === state.value.rsuIpv4) {
+      //     localMsgList.push(elem)
+      //     if (elem.type === 'srmTx') {
+      //       localSrmCount += 1
+      //     } else {
+      //       localSsmCount += 1
+      //     }
+      //   }
+      // }
+      // state.value.srmCount = localSrmCount
+      // state.value.srmSsmCount = localSsmCount
+      // state.value.srmMsgList = localMsgList
```

Comment:
Should this be commented out?

Reply:
This is not currently used, but might be on the future... Implemented!